torchgeo/transforms/intensity.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""TorchGeo intensity transforms."""
+
+import torch
+from kornia.augmentation import IntensityAugmentationBase2D
+from torch import Tensor
+
+
+class ToDecibelScale(IntensityAugmentationBase2D):
+    """Convert input tensor to decibel scale.
+
+    Primarily used for converting SAR pixel values to decibel scale.
+
+    .. versionadded:: 0.8
+    """
+
+    def __init__(
+        self, shift: float = 0.0, scale: float = 1.0, keepdim: bool = False
+    ) -> None:
+        """Initialize a new ToDecibelScale instance.
+
+        Args:
+            shift: The value to add to the decibel scale.
+                This is useful for avoiding negative values.
+            scale: An additional scale factor to apply to the decibel value.
+            keepdim: Whether to keep the output shape the same as input (True)
+                or broadcast it to the batch form (False).
+        """
+        super().__init__(p=1.0, p_batch=1.0, same_on_batch=True, keepdim=keepdim)
+        self.flags = {'scale': scale, 'shift': shift}
+
+    def apply_transform(
+        self,
+        input: Tensor,
+        params: dict[str, Tensor],
+        flags: dict[str, Tensor],
+        transform: Tensor | None = None,
+    ) -> Tensor:
+        """Apply the transform.
+
+        Args:
+            input: The input tensor.
+            params: Generated parameters.
+            flags: Static parameters.
+            transform: The geometric transformation tensor.
+
+        Returns:
+            The transformed input.
+        """
+        out = flags['scale'] * 10 * torch.log10(input + 1e-6) + flags['shift']
+        return out
+
+
+class Sentinel1ChangeMap(IntensityAugmentationBase2D):
```

Comment:
Not 100% sure if this should be public. Is it general purpose, or really specific to a single paper? Maybe we could add more description to the docstring explaining what it is for and how it works?

Reply:
It's pretty general imo. It's just an operation to compute the difference between 2 images and then threshold any change that's above some magnitude to be considered "change" vs "no change".

Reply:
I'll add some more details to the docstring.